src/App.tsx

```diff
@@ -17,18 +36,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                initialTodos={initialTodos}
+                setTodos={setTodos}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && todos?.length === 0 ? (
```

Comment:
```suggestion
              {loading && todos.length === 0 ? (
```
for example here will be no need to use optional chaining after fix initialTodos state